api/v1/verticadb_types.go

```diff
@@ -844,6 +844,23 @@ type VerticaDBStatus struct {
 	Sandboxes []SandboxStatus `json:"sandboxes,omitempty"`
 }
 
+type SandboxUpgradeState struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// UpgradeInProgress indicates if the sandbox is in the process
+	// of having its image change
+	UpgradeInProgress bool `json:"upgradeInProgress"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=status
+	// OfflineUpgradeInProgress indicates if the sandbox is in the middle
+	// of an offline upgrade
+	OfflineUpgradeInProgress bool `json:"offlineUpgradeInProgress"`
```

Comment:
Can we get away with just having a single bool for upgrade? Just the `UpgradeInProgress` one. I don't think we need to distinguish the different types of upgrade. Especially since we will only support one to start with.